src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs

```diff
@@ -37,6 +36,50 @@ public partial class Activity
         /// </summary>
         public string OperationName { get; }
 
+        /// <summary>
+        /// Is the ID of the current span. It is represented as an 8-byte array,
+        /// for example, 00f067aa0ba902b7
+        /// </summary>
+        public string SpanId { get; private set; }
+
+        /// <summary>
+        /// An 8-bit field that controls tracing flags such as sampling, trace level.
+        /// </summary>
+        public byte TraceFlags { get; set; } = 0;
+
+        internal bool Recorded
+        {
+            get => (TraceFlags & 1) == 1;
+            set => TraceFlags = (byte)(value ? 1 : 0); // TODO: TraceFlags could be 00000010, need to respect all bits
+        }
+        /// <summary>
+        /// Conveys information about request position in multiple distributed
+        /// tracing graphs and tracing-system specific context.
+        ///
+        /// Default .NET behavior is to blindly propagate tracestate to child activities and
+        /// outside the process. It's similar to Baggage, but cannot be set there because baggage
+        /// is sent in Correlation-Context and we can't make everyone update to newer .NET Core and new HttpClient.
+        ///
+        /// </summary>
+
+        //  The downside of this implementation (string) that
+        // Tracestate will be parsed and validated on each outgoing call.
+        //
```

Comment:
We should talk about the conditions where we have a tracestate and when we modify it.  

First I don't understand its purpose (the spec was frankly not much help, it just talks about its syntax, and what you are ALLOWED to do with it, not what its actually for).   From what I can tell it may just allow additional vendor-specific data to be passed along (what is its relationship with baggage?), which might include their vendor-specific ID.   From what I can tell however, I am just guessing around.   

Exactly when do we expect tracestate to even be present, and when do we expect it to be modified?    That will help significantly when thinking about the design.